File: php/tail.php
<?php
// Include the Postgres connection details
require 'connection.php';

// Begin a transaction
$connection->beginTransaction();
// Declare a cursor
$statement = $connection->prepare('DECLARE c CURSOR FOR TAIL demo');
// Execute the statement
$statement->execute();

/* Fetch all of the remaining rows in the result set */
while (true) {
    $tail = $connection->prepare('FETCH ALL c');
    $tail->execute();
    $result = $tail->fetchAll(PDO::FETCH_ASSOC);
    print_r($result);
}

File: rust/src/tail.rs
use std::{thread::sleep, time::Duration};

use crate::connection::create_client;

/// Run a tail over the PUBNUB materialized view
pub(crate) fn tail() {
    let mut client = create_client().expect("Error creating client.");
    let mut transaction = client.transaction().expect("Error creating transaction.");
    transaction.execute("DECLARE c CURSOR FOR TAIL (SELECT symbol, bid_price::text FROM market_orders) WITH (SNAPSHOT = false);", &[]).expect("Error creating cursor.");

    loop {
        let results = transaction.query("FETCH ALL c;", &[]).expect("Error running fetch.");
        for row in results {
            println!("{:} - {:}", row.get::<usize, String>(2), row.get::<usize, String>(3));
        }
        sleep(Duration::from_millis(200));
    }
}